py-rse/provenance.Rmd

```diff
@@ -0,0 +1,381 @@
+these analysis scripts and/or notebooks can simply be collected up into a zip folder
+for archiving.
+
+```shell
+$ zip scripts.zip bin/*.py
+```
+```text
+  adding: bin/collate.py (deflated 52%)
+  adding: bin/countwords.py (deflated 54%)
+  adding: bin/plotcounts.py (deflated 58%)
+  adding: bin/test_zipfs.py (deflated 67%)
+  adding: bin/utilities.py (deflated 51%)
+```
+
+### Data Processing Steps
+
+Finally,
+to make our results truly reproducible,
+the reader needs to know exactly how (and in what order) we ran our analysis scripts
+in our software environment.
+The way in which this information is collected and archived
+depends on the characteristics of the workflow.
+Our Zipf's Law analysis involved executing a series of command line programs,
+and a record of the command line entries we executed
+can be generated using our Makefile (Chapter \@ref(automate)).
+For instance, the sequence of commands involved in plotting
+the collated word count distribution is as follows:
+
+```shell
+$ make all -n -B
+```
+```text
+python bin/countwords.py data/dracula.txt > results/dracula.csv
+python bin/countwords.py data/time_machine.txt > results/time_machine.csv
+python bin/countwords.py data/sense_and_sensibility.txt > results/sense_and_sensibility.csv
+python bin/countwords.py data/sherlock_holmes.txt > results/sherlock_holmes.csv
+python bin/countwords.py data/moby_dick.txt > results/moby_dick.csv
+python bin/countwords.py data/frankenstein.txt > results/frankenstein.csv
+python bin/countwords.py data/jane_eyre.txt > results/jane_eyre.csv
+mkdir -p results
+python bin/collate.py results/time_machine.csv results/moby_dick.csv results/jane_eyre.csv results/dracula.csv results/sense_and_sensibility.csv results/sherlock_holmes.csv results/frankenstein.csv > results/collated.csv
+python bin/plotcounts.py results/collated.csv --outfile results/plotcounts.png --plotparams plotparams.yml
+```
+The `-n` option is for a dry run (i.e. the commands are displayed but not executed)
+and the `-B` option is to run all steps regardless or whether they need updating or not.
+
+We can archive these commands in a text file:
+
+```shell
+$ make all -n -B > commands.txt
+```
+
+Some of the parameters for `plotcounts.py` are in the `plotparams.yml` configuration file
+(Chapter \@ref(config)),
+so that file also needs to be archived.
+
+### Where to Archive Code
+
+Now that we have generated information to document our
+software environment (`environment.yml`),
+analysis scripts (`scripts.zip`),
+and data processing steps (`commands.txt` and `plotparams.yml`),
+we need to archive them somewhere
+(along with a README file to explain what each file is).
+There are a number online platforms dedicated to archiving (with a DOI)
+the "long-tail" of your research,
+including supplementary figures, data and/or code.
+Two of the most popular platforms for archiving code
+are [Zenodo][zenodo] and [Figshare][figshare].
```

Comment:
This can be linked with the above about Git tags, where a Git tag is versioned through the Zenodo-GitHub integration